crates/cli/src/parse/source.rs

```diff
@@ -113,3 +113,66 @@ pub(crate) fn parse_rpc_url(args: &Args) -> Result<String, ParseError> {
         Ok(url)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use thousands::Separable;
+
+    async fn setup_source(max_concurrent_requests: u64) -> Source {
```

Comment:
this is taken from `parse::timestamps` except I set the max_concurrent_requests to the input.